sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -81,9 +82,13 @@ object Cast {
     case (StringType, CalendarIntervalType) => true
     case (StringType, _: DayTimeIntervalType) => true
     case (StringType, _: YearMonthIntervalType) => true
+    case (_: IntegralType, DayTimeIntervalType(s, e)) if s == e => true
+    case (_: IntegralType, YearMonthIntervalType(s, e)) if s == e => true
 
     case (_: DayTimeIntervalType, _: DayTimeIntervalType) => true
     case (_: YearMonthIntervalType, _: YearMonthIntervalType) => true
+    case (DayTimeIntervalType(s, e), _: IntegralType) if s == e => true
+    case (YearMonthIntervalType(s, e), _: IntegralType) if s == e => true
```

Comment:
Since we can cast `INTERVAL DAY TO SECOND` to `INTERVAL SECOND`, I think it's a reasonable and convenient extension to cast `INTERVAL DAY TO SECOND` to integral type directly. That said, we can allow casting any interval type to integral type. For casting integral to interval, we should still only allow single field.

Reply:
> Since we can cast INTERVAL DAY TO SECOND to INTERVAL SECOND, I think it's a reasonable and convenient extension to > cast INTERVAL DAY TO SECOND to integral type directly.
So clever. You're absolutely right. I didn't think of this use case.